source/javascripts/floating-labels.coffee

```diff
@@ -0,0 +1,8 @@
+$('.floating-label .input').keyup ->
```

Comment:
Hey @chanhman,

Couple quick notes about this JS.

Repeatedly calling `$(this)`, `.closest()`, and `.find()`  is costly, as jQuery has to query the DOM for each of those calls.  It becomes especially worse when doing these calls from the `keyup` callback.  

A quick fix here would be to cache the elements you need prior to responding to `keyup`.

I've re-written the codepen you used into a jQuery plugin that does just that here:  https://codepen.io/anon/pen/jGrOJM

```javascript
$(document).ready(function() {
  $.fn.extend({
    floatingLabel: function(_options) {
      $(this).each(function() {
        $this = $(this);
        $input = $this.find("input");
        $label = $this.find("label, .placeholder");

        $input.keyup(function(e) {
          var hasValue = e.target.value.length;
          if (hasValue) {
            $label.addClass("is-visible active");
          } else {
            $label.removeClass("is-visible active");
          }
        });
      });
    }
  });

  $(".empty-val-input").floatingLabel();
});
```


You'll need to update the classnames to match yours (and feel free to convert to coffeescript)